ui-frontend/packages/catalog-ui-search/src/main/webapp/component/pages/metacard.tsx

```diff
@@ -3,33 +3,57 @@ import { hot } from 'react-hot-loader'
 import { useParams } from 'react-router-dom'
 import { useLazyResultsFromSelectionInterface } from '../selection-interface/hooks'
 import { GoldenLayout } from '../golden-layout/golden-layout'
-const Query = require('../../js/model/Query.js')
+import { DEFAULT_QUERY_OPTIONS, useUserQuery } from '../../js/model/TypedQuery'
+import {
+  FilterBuilderClass,
+  FilterClass,
+} from '../filter-builder/filter.structure'
+import { getFilterTreeForId } from './metacard-nav'
 const SelectionInterfaceModel = require('../selection-interface/selection-interface.model.js')
 const user = require('../singletons/user-instance.js')
 const _ = require('underscore')
 
-const getFilterTreeForId = ({ id }: { id: string }) => {
-  return {
-    type: 'AND',
-    filters: [
-      {
-        type: '=',
-        value: id,
-        property: '"id"',
-      },
-      {
-        type: 'ILIKE',
-        value: '*',
-        property: '"metacard-tags"',
-      },
-    ],
-  }
-}
+type UploadType = Backbone.Model<{
+  uploads: Backbone.Model<{ id: string; children: string[] }>[]
+}>
 
-const getQueryForId = ({ id }: { id: string }) => {
-  return new Query.Model({
-    filterTree: getFilterTreeForId({ id }),
-    federation: 'enterprise',
+export const getFilterTreeForUpload = ({
+  upload,
+}: {
+  upload: UploadType
+}): FilterBuilderClass => {
+  return new FilterBuilderClass({
+    type: 'OR',
+    filters: _.flatten(
+      upload
+        .get('uploads')
+        .filter((file) => file.id || file.get('children') !== undefined)
+        .map((file) => {
+          if (file.get('children') !== undefined) {
+            return file.get('children').map(
+              (child) =>
+                new FilterClass({
+                  type: '=',
+                  value: child,
+                  property: 'id',
+                })
+            )
+          } else {
+            return new FilterClass({
+              type: '=',
+              value: file.id,
+              property: 'id',
+            })
+          }
+        })
+        .concat(
+          new FilterClass({
+            type: '=',
+            value: '-1',
+            property: 'id',
+          })
+        )
```

Comment:
❓ So, if the file doesn't have any children, we default a filter with the file id and these lines. What does this do?

Reply:
I'm trying to remember why exactly, but it's been so long I can't.  From what I recall the queries didn't finish correctly without this.

However, at this point it might be a left over of an old limitation that is no longer necessary.